keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -242,16 +242,17 @@ class ApplicationService(
         version = version
       )
 
+    val pinnedArtifact = getPinnedArtifact(deliveryConfig, environmentName, artifact, version)
+
     return when (status) {
       PENDING -> {
-        val olderGitMetadata = repository.getGitMetadataByPromotionStatus(deliveryConfig, environmentName, artifact, CURRENT.name)
-
+        val olderGitMetadata = pinnedArtifact?: repository.getArtifactVersionByPromotionStatus(deliveryConfig, environmentName, artifact, CURRENT.name)
```

Comment:
So a pinned version doesn't get the status updated to `CURRENT`? That seems counter-intuitive...

Reply:
what do you mean? I didn't change the logic of the status updates, only for which version to compare against.

Reply:
If I'm reading this correctly, you'll give precedence to the pinned version instead of the version with a `CURRENT` status. I would expect that, whenever we deploy a version (regardless of whether it was as a result of a normal promotion or pinning), that version would become `CURRENT`. If that was the case, then the change in this specific line would be unnecessary because pinned == current.

Reply:
I see what you are saying. The current logic gives priority for pinned versions, and if we find one, we won't call to get the `current` version as well. I think we could have some edge cases where pinned != current, so I kinda think it's safer to keep it that way 🤷